replicate/common/generic-replicator.go

```diff
@@ -269,9 +269,11 @@ func (r *GenericReplicator) ResourceAdded(obj interface{}) {
 	if namespacePatterns, ok := annotations[ReplicateTo]; ok {
 		r.ReplicateToList[sourceKey] = struct{}{}
 
-		if list, err := r.Client.CoreV1().Namespaces().List(ctx, metav1.ListOptions{}); err != nil {
-			logger.WithError(err).Error("failed to list namespaces")
-		} else if err := r.replicateResourceToMatchingNamespaces(obj, namespacePatterns, list.Items); err != nil {
+		var namespaces []v1.Namespace
+		for _, ns := range namespaceWatcher.NamespaceStore.List() {
+			namespaces = append(namespaces, *ns.(*v1.Namespace))
+		}
```

Comment:
Minor optimization: Since the length of the namespace list is known beforehand, we can allocate the slice with the correct size from the start:

```suggestion
		namespacesFromStore := namespaceWatcher.NamespaceStore.List()
		namespaces := make([]v1.Namespace, len(namespacesFromStore))

		for i, ns := range namespacesFromStore {
			namespaces[i] = *ns.(*v1.Namespace)
		}
```

Reply:
Updates the commit and force-pushed :+1: